### PR TITLE
Adds new robit, to discourage facetanking.

### PR DESCRIPTION
**Pull request by Sebbe9123**
I forgot description.
Basically just adds a new robot with some tidbits of L O R E.
Bot has a special attack that injects a toxin which doesn't kill fast and is easily made a non-issue.
adds two whole iconfiles for mob.

### CODE REVIEW

**Comment by TheDavestDave** [comment]
Sorry what is this for? Exploration has spiders, deathclaw, and combat drones that easily kill anyone who stay still for long enough. Sometimes a few second lag is all it takes to get broken bones everywhere and IB. Combat drone can almost take on mechs. It might has uses in V2 but Mercs and combat drones are there too

**Comment by Novacat** [comment]
Intentions do not really matter since this PR does not add any spawn points for them. But, some pictures of the new mob would be nice.

**Comment by Wickedtemp** [comment]
It looks like its venom is zombie powder. That needs to be changed, because it's *not* a nonissue.

It won't do a whole lot of mechanical damage, but a 5u injection stuns AND mutes you for a whole 60 seconds, which is literally the same as executing you if it's a combat situation and you've got mobs still attacking. You can't move and you can't call for help.

**Comment by Heroman3003** [comment]
Current virgo mobs already have heavy facetanking discouragements in ways of dealing extremely high damage fast. Since after rework, facetanking already is among worst ideas.

File: code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+	var/poison_chance = 100
+	var/poison_per_bite = 5
+	var/poison_type = "zombiepowder"
+
+/datum/say_list/disbot
+	speak = list("ATTEMPTING TO CONTACT A.R.K, ATTEMPT 1e26+3","DIRT SAMPLE COLLECTED, DIRT QUOTA 124871/155 CONFIRMED.")
+	emote_see = list("Scans the dirt around it","Beeps as it scans a rock nearby")
+	say_maybe_target = list("BIOLOGICAL TRACES FOUND, ATTEMTPTING TO LOCATE SOURCE.","TRACE SOURCES FOUND, POWERING SCANNERS.",)
+	say_got_target = list("LIFEFORM LOCATED, ATTEMPTING TO COLLECT SAMPLE","CREATURE SPOTTED, PHERMONE GENERATORS DAMAGED, ATTEMPTING TO COLLECT GENETIC SAMPLE.")
```
Comment (Heroman3003) [inline code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm]: PHEROMONE perhaps?

File: code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	ai_holder_type = /datum/ai_holder/simple_mob/melee
+	say_list_type = /datum/say_list/disbot
+
+	var/poison_chance = 100
+	var/poison_per_bite = 5
+	var/poison_type = "zombiepowder"
+
+/datum/say_list/disbot
+	speak = list("ATTEMPTING TO CONTACT A.R.K, ATTEMPT 1e26+3","DIRT SAMPLE COLLECTED, DIRT QUOTA 124871/155 CONFIRMED.")
+	emote_see = list("Scans the dirt around it","Beeps as it scans a rock nearby")
```
Comment (Heroman3003) [inline code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm]: do not forget that emotes will be written as 
"Strange robot Scans the dirt around it" and such, so in emotes first letter needs to capitalization

File: code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	light_color = "#8a0707"
+	attacktext = "mauled"
+	attack_sound = 'sound/weapons/bite.ogg'
+
+	ai_holder_type = /datum/ai_holder/simple_mob/melee
+	say_list_type = /datum/say_list/disbot
+
+	var/poison_chance = 100
+	var/poison_per_bite = 5
+	var/poison_type = "zombiepowder"
```
Comment (Heroman3003) [inline code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm]: Zombie Powder is awful choice for an attack poison... Considering that it puts person out of commision for the longest time just for one hit. Nothing is less fun than having to wait several minutes after every encounter with one like that just for effects to wear off. And thats not counting that it will rapid-inject it more and more over time even after target is long down. It does not discourage facetanking, it encourages overprepping on carbon.

File: code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+	response_help = "pets the"
+	response_disarm = "gently pushes aside the"
+	response_harm = "hits the"
+	faction = "vagrant"
+	harm_intent_damage = 3
+	melee_damage_lower = 6
+	melee_damage_upper = 9
+	light_color = "#8a0707"
+	attacktext = "mauled"
```
Comment (Heroman3003) [inline code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm]: Robot is supposed to be trying to attack via wild injection stabs, no? Why 'mauled ' verb?

File: code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+	emote_see = list("Scans the dirt around it","Beeps as it scans a rock nearby")
+	say_maybe_target = list("BIOLOGICAL TRACES FOUND, ATTEMTPTING TO LOCATE SOURCE.","TRACE SOURCES FOUND, POWERING SCANNERS.",)
+	say_got_target = list("LIFEFORM LOCATED, ATTEMPTING TO COLLECT SAMPLE","CREATURE SPOTTED, PHERMONE GENERATORS DAMAGED, ATTEMPTING TO COLLECT GENETIC SAMPLE.")
+
+/mob/living/simple_mob/mechanical/infectionbot/apply_melee_effects(var/atom/A)
+	if(isliving(A))
+		var/mob/living/L = A
+		if(L.reagents)
+			var/target_zone = pick(BP_TORSO,BP_TORSO,BP_TORSO,BP_L_LEG,BP_R_LEG,BP_L_ARM,BP_R_ARM,BP_HEAD)
+			if(L.can_inject(src, null, target_zone))
```
Comment (Heroman3003) [inline code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm]: This basically means that wearing any kind of voidsuit or a set of armor with helmet makes you immune to the robot's attacks entirely, doesn't it? Considering can_inject, if memory fails me not, checks for coverage of the area. Once again, does not discourage facetanking in the slightest, just bringing more covering armor.

File: code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm (hunk 3 above)
Comment (Wickedtemp) [reply code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm]: Carbon won't do anything. There actually is no way to 'counter' this other than not getting jabbed in the first place. Regardless of the chems you stock up on beforehand, if you get jabbed, you're muted and stunned for 60 seconds for each time it hits you. If it doesn't stop hitting you, you're just kinda screwed and you're going to die.

File: code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm (hunk 5 above)
Comment (Sebbe9123) [reply code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm]: Non issue

File: code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm (hunk 4 above)
Comment (Sebbe9123) [reply code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm]: Oversight, I'll fix it.

File: code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm (hunk 1 above)
Comment (Sebbe9123) [reply code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm]: Whoops

**Comment by Wickedtemp** [comment]
So. Decided to see what'd happen if I went to fight a mob, but got "jabbed" by 5u zombie-powder.
https://gyazo.com/45ae484a2e637adda704106efb0c060e

TLDR: It hit me, I injected myself with the zombie powder to simulate a successful injection of venom. The rat quickly ate me, cue digestion, which would mean death due to my complete inability to move or call for help. I used OOC Escape, maybe some people will have mob-vore disabled. It continued to slash the downed character until they were in crit, at which point they slowly died.

So... Yeah. Pretty much went how I thought it was gonna go. Single inject basically gets you killed.

**Comment by Sebbe9123** [comment]
> Intentions do not really matter since this PR does not add any spawn points for them. But, some pictures of the new mob would be nice.
> 
> Also, since cataloguers is a thing now, I want a catalog entry for this mob.
> 
> PS: ...Is that a metroid?

Here's a gif of the two sprites: https://gyazo.com/cc1731274fc72c3f02d43d1c10751c6a
I didn't really notice before but they do look like metroids yea, I stole them from bay, pretty sure they are a construct for one of their cult modes.

File: code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm (hunk 2 above)
Comment (Sebbe9123) [reply code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm]: fixd

**Comment by Wickedtemp** [comment]
Sleep toxin is soporific, not zombie powder. Was the mob actually tested ingame at all? Does it attack once and leave you, or does it persist and continue to attack? Because if it's the latter, literally one successful jab will kill you due to your inability to call for help and fight back. Zombie powder is an instant-stun.

**Comment by Novacat** [comment]
I meant zombie powder, I wrote that review while incredibly sleepy.

**Comment by Sebbe9123** [comment]
> Going to want a catalog entry for this mob at bare minimum. I would be fine with the sleep toxin, but that will probably guarantee that this mob will be tagged at a high difficulty level and thus will be exceptionally rare.

Done

File: code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+	desc = "A strange and aged drone, this model appears to be made for gathering of genetic samples,\
+	sacrificing both power and durability for storage space and advanced scanners.\
+	The drone has clear dents and scratches all over it's casing bulging inwards in multiple areas,\
+	some even penetrated showing the advanced and headache inducing parts inside.\
+	<br><br>\
+	This model in particular has a collection of hostile and spinetingeling parts on the underside,\
+	small advanced anti-gravity generators located between giant syringes made for injecting and extracting all manners of fluids\
+	and samples of soil or other biological matter.\
+	Most interesting is the syringe it uses after taking a sample of any creature, the dry, flakey substance\
+	it injects has traces of bicardine and innaprovaline, now no more than rotten and expired toxins however.\
```
Comment (Novacat) [inline code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm]: ...Based on this description, I think it would be more lore fitting that, instead of zombie powder, the injected toxin is expired_medicine instead.

Comment (Sebbe9123) [reply code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm]: Isn't debilitating at all and makes the mob moat. I'm personally not going to do so.

Comment (Wickedtemp) [reply code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm]: Expired_medicine deals iirc 5 toxin per unit injected  5u injected per hit means 25 damage.

You can ignore it, but the thing about toxin is that if its left untreated and its around 50/60+, it will begin damaging your liver. Once your liver is dead, it moves on to other organs. It's either you fix it or you die. 

Did you test to see whether or not the mob continues to attack you while youre downed under the effects of zombie powder?

Comment (Novacat) [reply code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm]: If that is the case, than the mob's lore should be rewritten to compensate. Lore should fit what the mob is. It makes no sense that a super deadly toxin is coming from a mob that is... otherwise... fairly mundane in both catalog reward and description.

Comment (Sebbe9123) [reply code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm]: I doubt expired medicine is "dry, flakey"

Comment (Wickedtemp) [reply code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm]: It can be, if it's expired enough.

Comment (Novacat) [reply code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm]: As I said, not against the idea of an extremely dangerous mob, but, description should match the stats. If it is a murderdeathbot, it should be described as such. It is a bit of a stretch for what is essentially a malfunctioning probe to be so dangerous, especially since Zombie Powder shares pretty much none of the reagents of Inaprovaline or Bicaridine. There is no reason why those expired medicines would suddenly synthesize into Zombie Powder.

Comment (Sebbe9123) [reply code/modules/mob/living/simple_mob/subtypes/mechanical/disbot_vr.dm]: done
